lib/std/array_list.zig

```diff
@@ -162,15 +162,92 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
             self.items[n] = item;
         }
 
+        /// Resize the array, adding `count` new elements at position `index`, which have `undefined` values.
+        /// The return value is a slice pointing to the newly allocated elements. The returned pointer
+        /// becomes invalid when the list is resized. Resizes list if self.capacity is not large enough.
+        pub fn addManyAtIndex(
+            self: *Self,
+            index: usize,
+            count: usize,
+        ) Allocator.Error![]T {
```

Comment:
Could you give some example use cases for this function being `pub`? It also seems like this function could be removed entirely and the implementation could just be moved into `insertSlice` instead.

Reply:
I disagree here @squeek502 - if this logic is going to exist anyway, there's no reason not to expose it as a potentially-useful function. "Add new elements to the array which I'll initialize later" doesn't seem an unreasonable utility function at all - after all, this is why the existing `addMany*` functions exist (with this one being most analagous to `addManyAsSlice`).

Reply:
As an example, in my personal codebase I have an ArrayList of pixels. Data comes from files which have to be parsed and processed, then inserted into the ArrayList. It's better to ask the ArrayList to make space for the data and process the data directly into its final destination.